src/main/java/com/cedarsoftware/util/io/factory/ThrowableFactory.java

```diff
@@ -27,24 +29,70 @@
  */
 public class ThrowableFactory implements JsonReader.ClassFactory
 {
+    private static final String DETAIL_MESSAGE = "detailMessage";
+    private static final String CAUSE = "cause";
+    private static final String STACK_TRACE = "stackTrace";
+
     public Object newInstance(Class<?> c, JsonObject jsonObj)
     {
-        String msg = (String)jsonObj.get("detailMessage");
-        JsonObject jObjCause = (JsonObject)jsonObj.get("cause");
-        Throwable cause = null;
-        if (jObjCause != null)
-        {
-            JsonReader jr = new JsonReader();
-            cause = (Throwable) jr.jsonObjectsToJava(jObjCause);
+        String msg = (String) jsonObj.get(DETAIL_MESSAGE);
+        JsonObject causeMap = (JsonObject) jsonObj.get(CAUSE);
+
+
+        JsonReader reader = ReaderContext.instance().getReader();
+
+        Throwable cause = causeMap == null ? null : reader.convertParsedMapsToJava(causeMap, Throwable.class);
+        Throwable t = createException(c, msg, cause);
+
+        Object[] stackTraces = (Object[]) jsonObj.get(STACK_TRACE);
```

Comment:
What about setting the stackTrace to an empty[]?  I'm thinking from a security perspective, it provides value mainly to a hacker.